src/main/webapp/app/overview/submission-result-status.component.ts

```diff
@@ -48,7 +49,23 @@ export class SubmissionResultStatusComponent implements OnChanges {
             this.quizNotStarted = ArtemisQuizService.notStarted(quizExercise);
         } else {
             this.uninitialized = !afterDueDate && !this.studentParticipation;
-            this.notSubmitted = !!this.studentParticipation && !this.studentParticipation.submissions?.length;
+            this.notSubmitted = afterDueDate && !!this.studentParticipation && !this.studentParticipation.submissions?.length;
+        }
+
+        this.setShouldShowResult(afterDueDate);
+    }
+
+    private setShouldShowResult(afterDueDate: boolean) {
+        if (this.exercise.type === ExerciseType.QUIZ) {
+            this.shouldShowResult = !!this.studentParticipation?.results?.length;
+        } else if (this.exercise.type === ExerciseType.PROGRAMMING) {
+            const initializationStatesForResult = [InitializationState.INITIALIZED, InitializationState.INACTIVE, InitializationState.FINISHED];
```

Comment:
This should be `private static readonly`

Reply:
I don't really see the reason to create a variable here. What benefits would that bring?

Reply:
Let's move the discussion to the thread started by Julian: https://github.com/ls1intum/Artemis/pull/6835#discussion_r1257261162